scripts/py_dependency.py

```diff
@@ -0,0 +1,102 @@
+import sys
+from pathlib import Path
+from io import StringIO
+
+def add_to_dict(dictionary, key, val):
+        if key in dictionary:
+            dictionary[key].add(val)
+        else:
+            dictionary[key] = {val}
+
+def print_help():
+    print("dependency.py [OPTION]... [RECIPE]... TASK_DEPENDS")
+    print("dependency.py [OPTION]... [-t|--tree] RECIPE TASK_DEPENDS")
+    print("dependency.py [OPTION]... [-a|--all] TASK_DEPENDS")
+    print("SUMMARY:")
+    print("\tFind dependencies of recipes or packages")
+    print("\tRECIPE -> DEPENDENCY... - RECIPE needs DEPENDENCIES to be built")
+    print("\tRECIPE <- DEPENDANT... - --reverse: DEPENDANT needs RECIPE to be built")
+    print("\tTASK_DEPENDS can be piped, e.g.")
+    print("\tcat task-depends.dot | python dependency.py recipe")
+    print("OPTION:")
+    print("\t-h --help     - print this help")
+    print("\t-r --reverse  - reverse dependencies")
+    print("\t-t --tree     - prints all dependecies starting with RECIPE")
+    print("\t-a --all      - prints dependencies of all recipes(without duplicates).")
+    print("\t                Ignores RECIPE...")
+    print("\t              - -a and -t are mutually exclusive")
+    
+def replace_opt(opt):
+    opts = {
+        "--help": "h",
+        "--reverse": "r",
+        "--tree": "t",
+        "--all": "a",
+        }
+    if opt.startswith("--"):
+        return opts[opt] if opt in opts else ""
+    else:
+        return opt[1:]
+        
+
+def main():
+    opts = ''.join([replace_opt(opt) for opt in sys.argv[1:] if opt.startswith("-")])
+    args = [arg for arg in sys.argv[1:] if not arg.startswith("-")]
+    reverse = True if "r" in opts else False
+    print_all = True if "a" in opts else False
+    tree = True if "t" in opts else False
+    if "h" in opts or (tree and print_all) or not args:
+        print_help()
+        return
+    dependency_map = {}
+    if sys.stdin.isatty():
+        try:
+            input_string = Path(f"{args[-1]}", encoding="utf-8").read_text()
+        except Exception as ex:
+            print(ex)
```

Comment:
What is the purpose of catching an exception without error handling? When an exception is thrown it is printed. 
```
except Exception as ex:
    print(ex)
    return
```

This code does the same as this code:
```
```